src/yuzu/main.cpp

```diff
@@ -2840,170 +2840,335 @@ void GMainWindow::OnGameListNavigateToGamedbEntry(u64 program_id,
+                                     const std::filesystem::path& command,
+                                     const std::string& arguments, const std::string& categories,
+                                     const std::string& keywords, const std::string& name) {
+
+    // Copy characters if they are not illegal in Windows filenames
+    std::string filename = "";
+    const std::string illegal_chars = "<>:\"/\\|?*";
+    filename.reserve(name.size());
+    std::copy_if(name.begin(), name.end(), std::back_inserter(filename),
+                 [&illegal_chars](char c) { return illegal_chars.find(c) == std::string::npos; });
 
-    // If relative path, make it an absolute path
-    if (yuzu_command.c_str()[0] == '.') {
-        yuzu_command = Common::FS::GetCurrentDir() / yuzu_command;
+    if (filename.empty()) {
+        LOG_ERROR(Frontend, "Filename is empty");
+        return false;
     }
 
-#if defined(__linux__)
-    // Warn once if we are making a shortcut to a volatile AppImage
-    const std::string appimage_ending =
-        std::string(Common::g_scm_rev).substr(0, 9).append(".AppImage");
-    if (yuzu_command.string().ends_with(appimage_ending) &&
-        !UISettings::values.shortcut_already_warned) {
-        if (QMessageBox::warning(this, tr("Create Shortcut"),
-                                 tr("This will create a shortcut to the current AppImage. This may "
-                                    "not work well if you update. Continue?"),
-                                 QMessageBox::StandardButton::Ok |
-                                     QMessageBox::StandardButton::Cancel) ==
-            QMessageBox::StandardButton::Cancel) {
-            return;
-        }
-        UISettings::values.shortcut_already_warned = true;
+    if (!std::filesystem::is_regular_file(command)) {
+        LOG_ERROR(Frontend, "Command is not a regular file");
+        return false;
     }
-#endif // __linux__
 
-    std::filesystem::path target_directory{};
+    std::filesystem::path shortcut_path_full = shortcut_path / filename;
 
-    switch (target) {
-    case GameListShortcutTarget::Desktop: {
-        const QString desktop_path =
-            QStandardPaths::writableLocation(QStandardPaths::DesktopLocation);
-        target_directory = desktop_path.toUtf8().toStdString();
-        break;
+#if defined(__linux__) || defined(__FreeBSD__) // Linux and FreeBSD
+    // Reference for the desktop file template:
+    // https://specifications.freedesktop.org/desktop-entry-spec/desktop-entry-spec-1.0.html
+    try {
+
+        // Plus 'Type' is required
+        if (name.empty()) {
+            LOG_ERROR(Frontend, "Name is empty");
+            return false;
+        }
+
+        // Append .desktop extension
+        shortcut_path_full += ".desktop";
+
+        // Create shortcut file
+        std::ofstream shortcut_stream(shortcut_path_full, std::ios::binary | std::ios::trunc);
+
+        if (shortcut_stream.is_open()) {
```

Comment:
```suggestion
        if (!shortcut_stream.is_open()) {
            LOG_ERROR(Frontend, "Failed to create shortcut");
            return false;
        }
```

And then unindent the rest of the block